snap/snapcraft.yaml

```diff
@@ -1,98 +1,110 @@
 name: landscape-client
 base: core22
-version: '0.1'
+version: '23.08'
 icon: snap/gui/landscape-logo-256.png
 website: https://ubuntu.com/landscape
 summary: Client for the Canonical systems management product Landscape
 description: |
-  This client, when installed, allows a machine to connect to the
-  Landscape server and be remotely managed by it.  Be aware that
-  this snap is not a replacement for the apt package that is
-  traditionally used.  This snap is specifically designed for
-  coreXX based systems and as such can only interact with other
-  snaps, not apt packages.
+  Landcape is a web-based tool for managing Ubuntu systems. This snap, or the
+  equivalent debian package is necessary if you want your machine to be managed
+  in a Landscape account. It provides the Landscape client and requires a
+  Landscape account.
 
 grade: devel # must be 'stable' to release into candidate/stable channels
-architectures:
-  - build-on: amd64
-  - build-on: arm64
-  - build-on: ppc64el
-  - build-on: s390x
+architectures: [amd64, arm64, armhf, ppc64el, s390x]
 confinement: strict
 
-layout:
-  /var/lib/landscape:
-    bind: $SNAP_DATA/var/lib/landscape
-
-environment:
-  LD_LIBRARY_PATH: $LD_LIBRARY_PATH:$SNAP/usr/lib/x86_64-linux-gnu
-  PYTHONPATH: $SNAP/usr/lib/python3/dist-packages:$SNAP/usr/lib/python3.10/dist-packages
-
 apps:
   landscape-client:
     daemon: simple
-    command: bin/landscape-client
-    plugs:
-      - network
-  landscape-config:
-    command: bin/landscape-config
+    install-mode: disable
+    command: usr/bin/landscape-client
     plugs:
+      - hardware-observe
+      - mount-observe
       - network
+      - network-bind
+      - network-observe
+      - scsi-generic
+      - snapd-control
+      - system-observe
+    environment:
+      LANDSCAPE_CLIENT_SNAP: 1
+      LANDSCAPE_CLIENT_CORE: 1
+      PYTHONPATH: $SNAP/usr/lib/python3/dist-packages:$PYTHONPATH
+  config:
+    command: usr/bin/landscape-config
+    plugs: [network]
+    environment:
+      LANDSCAPE_CLIENT_SNAP: 1
+      LANDSCAPE_CLIENT_CORE: 1
+      PYTHONPATH: $SNAP/usr/lib/python3/dist-packages:$PYTHONPATH
+
+layout:
+  /etc/landscape-client.conf:
+    bind-file: $SNAP_COMMON/etc/landscape-client.conf
+  /var/lib/landscape/client:
+    bind: $SNAP_DATA/var/lib/landscape/client
```

Comment:
Where is SNAP_DATA? It might be weird if these files are buried in subdirectories where there's nothing else inside

Reply:
`$SNAP_DATA` is (currently) `/var/snap/landscape-client/<revision>` (https://snapcraft.io/docs/environment-variables#heading--snap-data). It's the suggested location for data that the snap uses that can be backed up and restored across `refresh`es and `revert`s. A lot of the snapcrafting docs examples suggest putting data files here, e.g., https://snapcraft.io/docs/snap-layouts.